pkg/ccl/backupccl/backup_test.go

```diff
@@ -10125,3 +10144,82 @@ func TestBackupRestoreSeparateExplicitIsDefault(t *testing.T) {
 		sqlDB.Exec(t, "DROP DATABASE inc_fkdb;")
 	}
 }
+
+// TestBackupNoOverwriteCheckpoint tests BACKUP to see that it no longer
+// overwrites the checkpoint file, while still correctly handling older
+// backups that do.
+func TestBackupNoOverwriteCheckpoint(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	defer func(oldInterval time.Duration) {
+		BackupCheckpointInterval = oldInterval
+	}(BackupCheckpointInterval)
+	BackupCheckpointInterval = 0
+
+	// We want the backup to be large enough so that it doesn't finish in
+	// one iteration, which wouldn't test if checkpoints overwrite.
+	const numAccounts = 1000
+	const userfile = "'userfile:///a'"
+	_, sqlDB, _, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
+	defer cleanupFn()
+
+	// Set the testing knobs so that userfile does not delete the filename
+	// is trying to write first. This means that if the file already exists
+	// it will throw an error when it tries to write. Make sure to set it
+	// back to true at the end of the test.
+	filetable.SetTestingKnobs(filetable.TestingKnobs{EnforceNoOverwrite: true})
+	defer filetable.SetTestingKnobs(filetable.TestingKnobs{EnforceNoOverwrite: false})
+	query := fmt.Sprintf("BACKUP TO %s", userfile)
+	sqlDB.Exec(t, query)
+}
+
+// TestBackupCheckpointInBaseDirectory tests to see that a checkpoint
+// file in the base directory can be properly read when one is not found
+// in progress. This can occur when an older version node creates the backup.
+func TestBackupCheckpointInBaseDirectory(t *testing.T) {
```

Comment:
Hmm I'm a little confused about what this test is trying to achieve. I get that an older version backup should write to the base directory, but an incremental backup does not really read the checkpoint file, right? It just operates on the previous backups manifest.

What I think we really want to test is a mixed version cluster running a backup, i.e. the node planning the backup is 21.2 but node executing is 22.1, and then node planning is 21.2 and executing is 22.1. This is much better suited for a roachtest like `backup/mixed-version` and I'd love to see that as a fast follow to this (and the LATEST) PR, wdyt?

Reply:
Oh duh, that sounds right that it doesn't even look at the checkpoints. I wrote something similar for latest files (which should work with this approach?) and just hastily copied it over assuming it would work here too. I think a roachtest would work best for this I agree. And although I don't think we necessarily need one for LATEST the logic will all be there so why not?